src/openms/source/ANALYSIS/ID/BasicProteinInferenceAlgorithm.cpp

```diff
@@ -265,50 +265,52 @@ namespace OpenMS
 
       auto current_best_pep_it = best_pep.find(lookup_seq);
       if (current_best_pep_it == best_pep.end())
-      {
-        auto &new_entry = best_pep[lookup_seq];
-        new_entry = std::map<Int, PeptideHit *>();
-        new_entry[lookup_charge] = &hit;
+      { // no entry exist for sequence? initialize seq->charge->&hit
+        best_pep[lookup_seq][lookup_charge] = &hit;        
```

Comment:
should work with value initialization @jpfeuffer or am I missing something

Reply:
I guess it would be even faster with emplace (only hashing once). Did not know about it back then.

Reply:
right could make a small difference